containers/src/Kokkos_DynamicView.hpp

```diff
@@ -173,7 +173,8 @@ struct ChunkedArrayManager {
     void execute() {
       // Destroy the array of chunk pointers.
       // Two entries beyond the max chunks are allocation counters.
-      for (unsigned i = 0; i < m_chunk_max; i++) {
+      size_t const len = *reinterpret_cast<size_t*>(m_chunks + m_chunk_max);
```

Comment:
```suggestion
      uintptr_t const len = *reinterpret_cast<uintptr_t*>(m_chunks + m_chunk_max);
```
just to be conforming with access in other locations. Also, I would very much prefer if the allocation counters are stored in separate variables (inside `ChunkedArrayManager`). The need to `reinterpret_cast` here feels pretty sketchy and is confusing regarding readability.

Reply:
Ok, I've made the change to `uintptr_t`

Reply:
I agree that allocating them separately would be increase code readability, but this PR is not making this worse (i.e., DynamicView has been this way since the beginning). Due to how critical this is for EMPIRE right now, I would like to get this moved forward as is and then push for refactoring later in improve code readability.